featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -328,14 +329,14 @@ def _run_dfs(self, entity, relationship_path, all_features, max_depth):
                                      max_depth=max_depth,
                                      relationship_path=sub_relationship_path)
 
-        """
-        Step 3 - Create Transform features
-        """
-        self._build_transform_features(
-            all_features, entity, max_depth=max_depth)
+        # add seed features, if any, for dfs to build on top of
+        for f in self.seed_features:
```

Comment:
any reason this can't stay inside of `_add_identity_features`?

Reply:
I think I extracted this because if seed features are added before building aggregations then this test fails:
https://github.com/Featuretools/featuretools/blob/4a620527c34a5f7a7484ed9f3a2f954eb7d1c1d8/featuretools/tests/synthesis/test_deep_feature_synthesis.py#L367-L378

But I'm realizing now that you could write a similar test without seed features and it would fail with the current implementation because I moved `_add_identity_features` before building aggregations:
```py
# We don't want this feature because it will always be equal to 'age'
assert not feature_with_name(features, 'LAST(sessions.customers.age)')
```

I need to think some more about how to solve this.

Reply:
I resolved this by adding identity and seed features first, and then making sure to not build aggregations of direct features of the target. For example, if we are building features for `customers` we do not use direct features such as `customers.age` as the base for aggregations.